columnflow/tasks/framework/plotting.py

```diff
@@ -178,21 +183,22 @@ class PlotBase1D(PlotBase):
         "default: False",
     )
 
-    def get_plot_parameters(self):
+    def get_plot_parameters(self) -> DotDict:
         # convert parameters to usable values during plotting
         params = super().get_plot_parameters()
 
-        params["skip_ratio"] = self.skip_ratio
-        params["yscale"] = None if self.yscale == law.NO_STR else self.yscale
-        params["shape_norm"] = self.shape_norm
-
+        params_to_add = {
+            "skip_ratio": self.skip_ratio,
+            "yscale": None if self.yscale == law.NO_STR else self.yscale,
+            "shape_norm": self.shape_norm,
+        }
+        params.cautious_update(params_to_add)
```

Comment:
See the last review comment first :)

The `cautious_update` can be dangerous here in case (e.g.) `yscale` is, say, already present and set to `None`. This would be fully consistent (since `None` would also be added through `params_to_add`), but still, `skip_ratio` and `shape_norm` would not be added to `params`.

Instead, maybe this is what one would like to read here as it is more explicit:

```python
... removed ...
```

... actually, I don't see a dependence among the three `params_to_add`, so I'm not sure why one would want to add all of them, but only if none of them is already present 🤔

Reply:
To handle the case where (part of) a `cautious_update` is consistent with the existing values one could explicitly check if these change (see above).

Reply:
Indeed. With the strict add method, this should be fine.